lucene/core/src/java/org/apache/lucene/search/IndriAndQuery.java

```diff
@@ -0,0 +1,22 @@
+package org.apache.lucene.search;
+
+import java.io.IOException;
+import java.util.List;
+
+/** A Query that matches documents matching combinations of 
+ * {@link TermQuery}s or other IndriAndQuerys.
+ */
+public class IndriAndQuery extends IndriQuery {
```

Comment:
Normally in Lucene `AND` implies `MUST`, i.e. required clauses.

But this query is actually disjunctive, right?  Documents will match even if they are missing some of the terms.  Should we name it `IndriOrQuery` maybe?  Or, `IndriBooleanQuery`?

Reply:
I agree the naming is confusing.  I have taken the naming schema as well as the logic from the original Indri search engine implementation.  The issue with renaming it is that there is already IndriOrQuery, which I have created and hope to be able to add at a future time.  I will continue to think about whether there is a better name for the IndriAndQuery though.

Reply:
OK let's keep the (somewhat confusing) naming for now.  Naming is the hardest part!!